core-bundle/src/EventListener/DataContainer/PurgeMemberRegistrationsListener.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\Framework\ContaoFramework;
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Contao\CoreBundle\ServiceAnnotation\Callback;
+use Contao\MemberModel;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+/**
+ * Purges the member registrations in the front and back end, whenever tl_member is loaded (#3711).
+ *
+ * @internal
+ *
+ * @Callback(table="tl_member", target="config.onload")
+ */
+class PurgeMemberRegistrationsListener
```

Comment:
We should use the `reviseTable` hook instead of the `onload_callback`. `reviseTable` is only executed in exactly the case where this should be done, and it supports reloading the UI if something was deleted (dunno if that's actually necessary 🤷 )

Reply:
Should we use `onload` only for the front end then, so that it is executed for any module related to `tl_member`?

Reply:
why do we need to purge in the front end? To make sure a user can register again? maybe we could explicitly check that on registration?

Reply:
> why do we need to purge in the front end? To make sure a user can register again?

Yes, otherwise this wouldn't fix the problem.

> maybe we could explicitly check that on registration?

Yes, I guess it's sufficient to explicitly do it in `ModuleRegistration` only.